commands/deploy.go

```diff
@@ -250,3 +252,45 @@ func triggerUpdate(accountID, appID int, payloadID, serviceURL string, c *http.C
 	}
 	return nil
 }
+
+// newFileUploadRequest builds a HTTP request for uploading an app and the account + app it belongs to
+func newFileUploadRequest(c *DeployCmd, f *os.File) (r *http.Request, err error) {
+	contents, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	body := new(bytes.Buffer)
```

Comment:
(nitpick) I prefer `var body bytes.Buffer`. No need for `new` here.

Reply:
Good call, this was a copypasta from a gist 😂

Reply:
So fun little thing here: 

The `new` allocates zero memory to the variable. 

The more idiomatic `var body *bytes.Buffer` causes a panic in `multipart.NewWriter(body)` because it's a nil pointer dereference.

I'm gonna change this line to be `body := &bytes.Buffer{}` so the code style stays consistent by eliminating the `new` and still not cause `multipart.NewWriter()` to panic.

Reply:
No, you are supposed to do this:

```
var body bytes.Buffer
```

then use `&body` as a parameter to the func.